packages/search-ui-site-search-connector/src/SiteSearchAPIConnector.js

```diff
@@ -36,7 +31,20 @@ function _request(engineKey, method, path, params) {
     }),
     credentials: "include"
   }).then(response => {
-    return response.json();
+    if (response.status === 200) {
+      return response.json();
+    } else {
+      return response
+        .json()
+        .then(json => {
```

Comment:
This seems weird... seems like the `.then()` block throws an error, which will always be caught by the `.catch()` block, which will ignore the original `message` content of the Error from the `.then()` block. AFAICT this would NEVER throw an Error with `json.error` as the Error's argument.  It might be clearer to write this out using async/await.